providers/terraform/provider/detector.go

```diff
@@ -48,26 +50,37 @@ func (s *Service) detect(asset *inventory.Asset, conn *connection.Connection) er
 	}
 	asset.Platform = p
 
-	projectPath := asset.Connections[0].Options["path"]
-	absPath, _ := filepath.Abs(projectPath)
-	h := sha256.New()
-	h.Write([]byte(absPath))
-	hash := hex.EncodeToString(h.Sum(nil))
-	platformID := "//platformid.api.mondoo.app/runtime/terraform/hash/" + hash
-	asset.Connections[0].PlatformId = platformID
-	asset.PlatformIds = []string{platformID}
+	// we always prefer the git url since it is more reliable
+	url, ok := asset.Connections[0].Options["ssh-url"]
+	if ok {
+		domain, org, repo, err := parseSSHURL(url)
+		if err != nil {
+			return err
+		}
+		platformID := "//platformid.api.mondoo.app/runtime/terraform/domain/" + domain + "/org/" + org + "/repo/" + repo
+		asset.Connections[0].PlatformId = platformID
+		asset.PlatformIds = []string{platformID}
```

Comment:
could we add both platform ids to the platform ids array? so that if someone scans terraform directly and via gitlab they would come up with the same asset

Reply:
Yes we should, lets do this in a followup PR.